plugins/modules/site_workflow_manager.py

```diff
@@ -482,29 +575,140 @@ def site_exists(self):
+                    self.log("Site '{0}' exists in Cisco Catalyst Center".format(self.want.get("site_name")), "INFO")
+                else:
+                    self.log("No valid site details found for '{0}'".format(self.want.get("site_name")), "WARNING")
+            return (site_exists, current_site)
+
+        else:
+            self.get_ccc_version_as_integer() >= self.version_2_3_7_6
+            try:
+                response = {}
+                bulk_operation = self.config[0].get('site', {}).get('bulk_operation', False) or self.config[0]['site']['type']
+                self.log(bulk_operation)
+                if bulk_operation:
+                    name_list = [site.get("name") for site in self.want if "name" in site]
+                    self.log(name_list)
+                    all_sites_info = []
+                    for name in name_list:
+                        try:
+                            response = self.get_site(name)
+                            response_data = response.get("response", [])
+
+                            if not response_data:
+                                self.log("No site information found for name: {0}".format(name), "WARNING")
+                                return (site_exists, current_site)
+
+                            for site in response_data:
+                                if isinstance(site, dict):
+                                    current_site = {}
+                                    for key, value in site.items():
+                                        current_site[key] = value
+                                    if site.get('nameHierarchy'):
+                                        name_hierarchy = site.get('nameHierarchy', '')
+                                        current_site['parentName'] = name_hierarchy.rsplit('/', 1)[0]
+                                    else:
+                                        current_site['parentName'] = None
+
+                                    all_sites_info.append(current_site)
+                                    site_exists = True
+                        except Exception as e:
+                            self.log("Error fetching site for name '{0}': {1}".format(name, str(e)))
+
+                    return (site_exists, all_sites_info)
+
+            except Exception as e:
+                self.log("Bulk operation is not available due to : {0}".format(str(e)), "WARNING")
+                name_hierarchy = self.want.get("site_name")
+                try:
+                    response = self.get_site(name_hierarchy)
+                except Exception as e:
+                    self.log("Error fetching site for {0}: {1}".format(name_hierarchy, str(e)))
+                response = response.get("response")
+                if not response:
+                    self.log("No site information found for name hierarchy: {0}".format(name_hierarchy), "WARNING")
+                    return (site_exists, current_site)
+                if response:
```

Comment:
no need to check response again. Already checked in 646..

Reply:
Removed